src/brevitas_examples/common/learned_round/learned_round_method.py

```diff
@@ -0,0 +1,226 @@
+
+        rec_loss = F.mse_loss(pred, tgt, reduction='none').sum(1).mean()
+
+        if self.iter < self.loss_start:
+            b = self.temp_decay(self.iter)
+            round_loss = 0
+        else:  # 1 - |(h-0.5)*2|**b
+            b = self.temp_decay(self.iter)
+            round_vals = self.learned_round_module.p_forward()
+            round_loss = self.weight * (1 - ((round_vals - 0.5).abs() * 2).pow(b)).sum()
+
+        total_loss = rec_loss + round_loss
+        return total_loss, (total_loss, rec_loss, round_loss, b)
+
+    def format_loss_components(self, loss: float, rec_loss: float, round_loss: float, b) -> str:
+        return "loss = {:.4f}, rec_loss = {:.4f}, round_loss = {:.4f}, b = {:.4f}".format(
+            loss, rec_loss, round_loss, b)
+
+
+class AdaRound(LearnedRound):
+
+    def __init__(
+            self,
+            iters: int = 200,
+            weight: float = 0.01,
+            b_range: Tuple = (20, 2),
+            warmup: float = 0.2,
+            decay_start: float = 0.0,
+            **kwargs,
+    ) -> None:
+        super().__init__(iters, **kwargs)
+        # Loss-related configuration
+        self.weight = weight
+        self.b_range = b_range
+        self.warmup = warmup
+        self.decay_start = decay_start
+
+    def _is_learned_round_module(self, module: nn.Module) -> bool:
+        return isinstance(module, LearnedRoundSte)
+
+    def _insert_learned_round_quantizer_to_layer(
+            self,
+            layer: nn.Module,
+            learned_round_zeta: float = 1.1,
+            learned_round_gamma: float = -0.1) -> None:
+        floor_weight = torch.floor(layer.weight.data / layer.quant_weight().scale)
+        delta = (layer.weight.data / layer.quant_weight().scale) - floor_weight
+        value = -torch.log((learned_round_zeta - learned_round_gamma) /
+                           (delta - learned_round_gamma) - 1)
+        layer.weight_quant.quant_injector = layer.weight_quant.quant_injector.let(
+            float_to_int_impl_type=FloatToIntImplType.LEARNED_ROUND,
+            learned_round_impl_type=LearnedRoundImplType.HARD_SIGMOID,
+            learned_round_gamma=learned_round_gamma,
+            learned_round_zeta=learned_round_zeta,
+            learned_round_init=value)
+
+    def _instantiate_loss(
+            self, block: nn.Module, learned_round_modules: List[nn.Module]) -> AdaRoundLoss:
+        return AdaRoundLoss(
+            block,
+            learned_round_modules,
+            max_count=self.iters,
+            weight=self.weight,
+            warmup=self.warmup,
+            decay_start=self.decay_start,
+        )
+
+
+class AutoRoundLoss(LearnedRoundLoss):
+
+    def __call__(self, pred: torch.Tensor, tgt: torch.Tensor) -> Tuple[torch.Tensor, Tuple]:
+        loss = F.mse_loss(pred, tgt)
+        return loss, (loss,)
+
+    def format_loss_components(self, loss: float) -> str:
+        return "loss = {:.4f}".format(loss)
+
+
+class AutoRound(LearnedRound):
```

Comment:
Same as above, rename to something that makes more clear the difference between this FloatToIntImpl and the one above.